extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMusicTrendingExtractor.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Created by Christian Schabesberger on 12.08.17.
+ *
+ * Copyright (C) Christian Schabesberger 2018 <chris.schabesberger@mailbox.org>
+ * YoutubeTrendingExtractor.java is part of NewPipe Extractor.
+ *
+ * NewPipe Extractor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * NewPipe Extractor is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with NewPipe Extractor. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.schabi.newpipe.extractor.services.youtube.extractors;
+
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getJsonPostResponse;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getTextAtKey;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.prepareDesktopJsonBuilder;
+import static org.schabi.newpipe.extractor.utils.Utils.UTF_8;
+import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
+
+import com.grack.nanojson.JsonObject;
+import com.grack.nanojson.JsonWriter;
+
+import org.schabi.newpipe.extractor.Page;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.kiosk.KioskExtractor;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+import org.schabi.newpipe.extractor.localization.TimeAgoParser;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+
+public class YoutubeMusicTrendingExtractor extends KioskExtractor<StreamInfoItem> {
+    private JsonObject initialData;
+
+    public YoutubeMusicTrendingExtractor(final StreamingService service,
+                                         final ListLinkHandler linkHandler,
+                                         final String kioskId) {
+        super(service, linkHandler, kioskId);
+    }
+
+    @Override
+    public void onFetchPage(@Nonnull final Downloader downloader)
+            throws IOException, ExtractionException {
+        // @formatter:off
+        final byte[] body = JsonWriter.string(prepareDesktopJsonBuilder(getExtractorLocalization(),
+                getExtractorContentCountry())
+                .value("browseId", "MUtrending")
```

Comment:
I tried looking for this `browseId` everywhere, on `youtube.com` and `music.youtube.com` (There's no trending page at all).

What I found out is that the official client uses a `browseId` of `FEtrending` with the `params` as `4gINGgt5dG1hX2NoYXJ0cw%3D%3D`, which is URL and base64 encoded protobuf.

![image](https://user-images.githubusercontent.com/20838718/198099614-e60e2716-6d58-4134-8ee7-9634cf1014ae.png)

The protobuf decoded looks like this:
![image](https://user-images.githubusercontent.com/20838718/198097567-2bfe2e9d-b208-44cc-9993-8e9633417b81.png)

I think we should change this to follow the official client.

Reply:
music: '4gINGgt5dG1hX2NoYXJ0cw%3D%3D',
  gaming: '4gIcGhpnYW1pbmdfY29ycHVzX21vc3RfcG9wdWxhcg%3D%3D',
  movies: '4gIKGgh0cmFpbGVycw%3D%3D'